### PR TITLE
FW: implement negated --cpuset option

### PR DESCRIPTION
**Pull request by thiagomacieira**
This is useful when you want to run on all logical processors except a few, and listing all 896 entries but those would be very tiring.

For example, for `--cpuset='!0,3'` on a 4-core system:
```yaml
cpu-info:
  0: { logical: 4, package: 0, core: 0, thread: 1, family: 6, model: 0x8c, stepping: 1, microcode: 0xb6, ppin: null }
  1: { logical: 1, package: 0, core: 1, thread: 0, family: 6, model: 0x8c, stepping: 1, microcode: 0xb6, ppin: null }
  2: { logical: 5, package: 0, core: 1, thread: 1, family: 6, model: 0x8c, stepping: 1, microcode: 0xb6, ppin: null }
  3: { logical: 2, package: 0, core: 2, thread: 0, family: 6, model: 0x8c, stepping: 1, microcode: 0xb6, ppin: null }
  4: { logical: 6, package: 0, core: 2, thread: 1, family: 6, model: 0x8c, stepping: 1, microcode: 0xb6, ppin: null }
  5: { logical: 7, package: 0, core: 3, thread: 1, family: 6, model: 0x8c, stepping: 1, microcode: 0xb6, ppin: null }
```

And with `--cpuset='!c2'`:
```yaml
cpu-info:
  0: { logical: 0, package: 0, core: 0, thread: 0, family: 6, model: 0x8c, stepping: 1, microcode: 0xb6, ppin: null }
  1: { logical: 4, package: 0, core: 0, thread: 1, family: 6, model: 0x8c, stepping: 1, microcode: 0xb6, ppin: null }
  2: { logical: 1, package: 0, core: 1, thread: 0, family: 6, model: 0x8c, stepping: 1, microcode: 0xb6, ppin: null }
  3: { logical: 5, package: 0, core: 1, thread: 1, family: 6, model: 0x8c, stepping: 1, microcode: 0xb6, ppin: null }
  4: { logical: 3, package: 0, core: 3, thread: 0, family: 6, model: 0x8c, stepping: 1, microcode: 0xb6, ppin: null }
  5: { logical: 7, package: 0, core: 3, thread: 1, family: 6, model: 0x8c, stepping: 1, microcode: 0xb6, ppin: null }
```

[ChangeLog][Framework] Added the ability to negate the selection of logical processors using the the `--cpuset` option: if the first character in that option's argument is an exclamation mark (!), then the argument removes the matched entries from the list. This option can be combined with other `--cpuset` options to add to or remove from the working set.

### CODE REVIEW

File: framework/topology.cpp [hunk 1]
```diff
@@ -774,9 +774,23 @@ void apply_cpuset_param(char *param)
...
+        // we're removing from the existing set
+        new_cpu_info = { old_cpu_info.begin(), old_cpu_info.end() };
+        add = false;
+        ++param;
+    }
+
     std::string p = param;
     for (char *arg = strtok(p.data(), ","); arg; arg = strtok(nullptr, ",")) {
         const char *orig_arg = arg;
+        struct MatchCpuInfoByCpuNumber {
```
Comment (busykai) [inline framework/topology.cpp:788]: struct declaration in a loop looks weird, would be better to move it out or event to the beginning of the function.

File: framework/topology.cpp [hunk 2]
```diff
@@ -796,12 +810,21 @@ void apply_cpuset_param(char *param)
         };
         auto add_to_set = [&](const struct cpu_info &cpu) {
```
Comment (busykai) [inline framework/topology.cpp:811]: this is no longer `add_to_set`. maybe `modify_set`?

File: framework/topology.cpp (hunk 1 above)
Comment (thiagomacieira) [reply framework/topology.cpp:788]: Moved.

File: framework/topology.cpp (hunk 2 above)
Comment (thiagomacieira) [reply framework/topology.cpp:811]: I thought I'd renamed it to `apply_to_set`. Looks like I missed something.

**Pull request merged**